### DISCUSSION TITLE
yaml linter `spectral` spawns the locally installed matrix client named `spectral`

### DISCUSSION THREAD

**Discussion by sumpfralle** [w0rp/ale#4143] [Q&A - Ask for help with problems] [answered]
I am using Debian and have the Matrix desktop client [spectral](https://gitlab.com/b0/spectral) installed (as a Debian package).
ALE is also installed as a Debian package (`vim-ale`) in version 3.1.0.

When opening a yaml file with vim, the `spectral` matrix client (`/usr/bin/spectral`) is spawn again and again (instead of the wanted yaml linter, which is not installed).

The corresponding linter description file in the Debian package is unchanged (see [ale_linters/yaml/spectral.vim](https://github.com/dense-analysis/ale/blob/master/ale_linters/yaml/spectral.vim)).

Does someone have a hint, how I should change the linter description in order to prevent `/usr/bin/spectral` from being used?
(I would report this to the Debian package maintainer, if such a change should not be suitable for `vim-ale` itself)

**Comment by hsanson** [comment, marked as answer]
You can set the executable variable to something that does not exists so ALE fails to find the binary and won't invoke the linter:

```
let g:ale_yaml_spectral_executable = '/nonexists'
```

**Reply by sumpfralle** [reply]
Thank you!

I placed this line into my `.vimrc` and the process spawning came to an end.

I would have preferred an approach to "fix" the detection. But this seems to be really impossible, since `spectral` (the Matrix desktop client) simply does not evaluate any command line arguments. Thus there is no way to distinguish the real `spectral` linter from the Matrix desktop client.
